sphinx/directives/other.py

```diff
@@ -130,6 +131,11 @@ def parse_content(self, toctree: addnodes.toctree) -> List[Node]:
                                    location=toctree)
                     self.env.note_reread()
                 else:
+                    if not included(self.env.doc2path(docname, False)):
```

Comment:
If my understanding is correct, the "not included" documents are all ignored on `Project.discover()`. As a result, they're not added to `env.found_docs`. If so, they will never be reached here because former if-block catches them.